go/enclave/storage/storage.go

```diff
@@ -303,39 +304,44 @@ func (s *storageImpl) FetchBatchesByBlock(block common.L1BlockHash) ([]*core.Bat
 	return enclavedb.ReadBatchesByBlock(s.db.GetSQLDB(), block)
 }
 
-func (s *storageImpl) StoreBatch(batch *core.Batch, receipts []*types.Receipt) error {
+func (s *storageImpl) StoreBatch(batch *core.Batch) error {
 	// sanity check that this is not overlapping
 	prev, err := s.FetchBatchBySeqNo(batch.SeqNo().Uint64())
-	if err == nil && !bytes.Equal(prev.Hash().Bytes(), batch.Hash().Bytes()) {
+	if err == nil && prev.Hash() != batch.Hash() {
+		s.logger.Crit(fmt.Sprintf("Conflicting batches for the same sequence %d: (previous) %s != (incoming) %s", batch.SeqNo(), prev.Hash(), batch.Hash()))
 		return fmt.Errorf("a different batch with same sequence number already exists: %d", batch.SeqNo())
 	}
 
+	if prev != nil {
```

Comment:
Could do with a comment to explain what case this is covering. I guess it's like `// duplicate - already processed batch with this seq number and hash`

Reply:
Might be slightly more intuitive to call the batch with the same seq num `existing` rather than `prev`